indico/modules/events/registration/google_wallet.py

```diff
@@ -0,0 +1,265 @@
+                    'language': 'en-US',
+                    'body': make_format_event_date_func(self.event.category)(self.event),
+                    'id': 'datefield'
+                },
+            ],
+            'classTemplateInfo': {
+                'cardTemplateOverride': {
+                    'cardRowTemplateInfos': [{
+                        'oneItem': {
+                            'item': {
+                                'firstValue': {
+                                    'fields': [{
+                                        'fieldPath': "class.textModulesData['datefield']"
+                                    }]
+                                }
+                            },
+                        }
+                    }, {
+                        'twoItems': {
+                            'startItem': {
+                                'firstValue': {
+                                    'fields': [{
+                                        'fieldPath': "object.textModulesData['namefield']"
+                                    }]
+                                }
+                            },
+                            'endItem': {
+                                'firstValue': {
+                                    'fields': [{
+                                        'fieldPath': "object.textModulesData['emailfield']"
+                                    }]
+                                }
+                            },
+                        }
+                    }]
+                }
+            }
+        }
+        if not dict_template['venue']['name']['defaultValue']['value']:
+            dict_template.pop('venue')
+        if self.event.has_logo and self.event.is_public:  # Event logo MUST be reachable via URL as anon
+            dict_template['heroImage'] = {
+                'sourceUri': {
+                    'uri': self.event.external_logo_url
+                }
+            }
+        signals.event.google_wallet_class_template_created.send(self.event, data=dict_template)
+        return dict_template
+
+    def create_class(self, issuer_id: str, class_suffix: str, update: bool = False) -> str:
+        """Create/update a class."""
+        # Check if the class exists
+        response = self.http_client.get(url=f'{self.class_url}/{issuer_id}.{class_suffix}')
+
+        if response.status_code == 200:
+            # Class already exists
+            if update:
+                updated_class = self.create_class_template(issuer_id, class_suffix)
+                response = self.http_client.put(url=f'{self.class_url}/{issuer_id}.{class_suffix}', json=updated_class)
+                if response.status_code != 200:
+                    raise BadRequest(response.text)  # Something else went wrong...
+            return response.json()
+        elif response.status_code != 404:
+            Logger.get('events').warning('Cannot create Google Wallet class: %s', response.text)
```

Comment:
This error message is misleading, it's getting the existing class that failed, not creating a new one.